### DISCUSSION TITLE
Jupyter with local neurodesktop

### DISCUSSION THREAD

**Discussion by JonteP** [neurodesk/neurodesk.github.io-hugo#265] [General]
I am curious whether it is possible to interface with neurodesk via jupyter notebooks when running it locally in docker in the same way as through lab/play? 
The closest I was able to get was running jupyter in VSCode within neurodesktop, but it doesn't seem to provide the same functionality.

**Comment by aswinnarayanan** [comment]
Hi @JonteP. We're currently finalising a new version of neurodesktop that's fully based on the jupyterlab environment just like lab/play . With the release of this version, the local, lab and play environments will all be using the same container.
https://github.com/NeuroDesk/neurodesktop/tree/jupyter-rebase
It should be just a week or two away from release, just doing the final tweaks and testing at the moment.

**Reply by JonteP** [reply]
This is great news, thank you!

**Reply by JonteP** [reply]
> Hi @JonteP. We're currently finalising a new version of neurodesktop that's fully based on the jupyterlab environment just like lab/play . With the release of this version, the local, lab and play environments will all be using the same container. https://github.com/NeuroDesk/neurodesktop/tree/jupyter-rebase It should be just a week or two away from release, just doing the final tweaks and testing at the moment.

Just tried out the new Neurodesk App and it looks great! Is there a way to mount local folders and samba shares? With neurodesktop, I simultaneously mount a local folder and our data storage server, the latter by adding a samba share as a volume in docker and mounting that when launching the neurodesktop docker image.

With Neurodesk App it seems that C:\Users\[username]\neurodesktop-storage is always mounted.

**Reply by iishiishii** [reply]
Hi @JonteP 
Thank you for your feedback and interest in the new feature for our app! 
We are working on the development of this feature, but there will be a slight delay in its release.
We apologize for any inconvenience because of this delay and will keep you updated on the progress.
Cheers.

**Reply by JonteP** [reply]
> Hi @JonteP Thank you for your feedback and interest in the new feature for our app! We are working on the development of this feature, but there will be a slight delay in its release. We apologize for any inconvenience because of this delay and will keep you updated on the progress. Cheers.

Ok, thank you! Glad to know it's being worked on. 🙂

**Reply by JonteP** [reply]
Ok, so I tried working with Jupyter notebooks in the new NeurodeskApp. I used the example notebooks here as a starting point, however, I cannot figure out how to load modules. 
`import lmod`
`await lmod.load('[modulename]')`

gives me the error: "TypeError: object NoneType can't be used in 'await' expression".

**Reply by stebo85** [reply]
Dear @JonteP,

thank you for finding this issue - we haven't completed the module load setup for notebooks, but this should only take a few days to finish up.

thank you
steffen

**Reply by stebo85** [reply]
I made progress on this. This is actually a bug in jupyterlmod and was recently introduced: https://github.com/cmd-ntrf/jupyter-lmod/issues/63 - I will try to revert back to an older version of jupyterlmod until they fixed it.

**Reply by stebo85** [reply]
I reverted back to jupyterlmod 3.0.0; @aswinnarayanan - are the jupyter-rebase builds automatically build and released already?

**Reply by iishiishii** [reply]
Hi @JonteP 
The custom data storage feature is now part of the newest version of NeurodeskApp. Thanks for being patient with us.
You can first uninstall the old version and grab the latest version for your operating system right [here](https://github.com/NeuroDesk/neurodesk-app/releases/tag/v0.2.0).
To use custom data directory, you can follow [this instruction](https://github.com/NeuroDesk/neurodesk-app/tree/master#add-custom-data-directory).
Please let us know if you find other issues.
Cheers.

**Reply by JonteP** [reply]
Thank you for implementing this!
